src/sidebar/components/test/tag-editor-test.js

```diff
@@ -1,63 +1,290 @@
+      'tag-editor-datalist-annotation_id'
+    );
+  });
+
+  it('generates a ordered datalist containing the array values returned from fakeTagsService.filter ', () => {
+    const wrapper = createComponent();
+    // simulate `keyup` to populate suggestions list
+    wrapper.find('input').instance().value = 'non-empty';
+    wrapper.find('input').simulate('input', { inputType: 'insertText' });
+
+    // fakeTagsService.filter returns ['tag4', 'tag3'], but
+    // datalist shall be ordered as ['tag3', 'tag4']
+    assert.equal(
+      wrapper
+        .find('datalist option')
+        .at(0)
+        .prop('value'),
+      'tag3'
+    );
+    assert.equal(
+      wrapper
+        .find('datalist option')
+        .at(1)
+        .prop('value'),
+      'tag4'
+    );
+  });
+
+  [
+    {
+      text: ' in Chrome and Safari',
+      eventPayload: { inputType: undefined },
+    },
+    {
+      text: ' in Firefox',
+      eventPayload: { inputType: 'insertReplacementText' },
+    },
+  ].forEach(test => {
+    it(`clears the suggestions when selecting a tag from datalist ${test.text}`, () => {
+      const wrapper = createComponent();
+      wrapper.find('input').instance().value = 'tag3';
+      typeInput(wrapper);
+      assert.equal(wrapper.find('datalist option').length, 2);
+      wrapper.find('input').simulate('input', test.eventPayload); // simulates a selection
+      assert.equal(wrapper.find('datalist option').length, 0);
+
+      assert.isTrue(
+        fakeTagsService.store.calledWith([
+          { text: 'tag1' },
+          { text: 'tag2' },
+          { text: 'tag3' },
+        ])
+      );
     });
-    assert.deepEqual(element.ctrl.tagList, [{ text: 'foo' }, { text: 'bar' }]);
   });
 
-  describe('when tags are changed', function() {
-    let element;
-    let onEditTags;
+  it('clears the suggestions when deleting input', () => {
+    const wrapper = createComponent();
+    wrapper.find('input').instance().value = 'tag3';
+    typeInput(wrapper);
+    assert.equal(wrapper.find('datalist option').length, 2);
+    wrapper.find('input').instance().value = '';
+    wrapper
+      .find('input')
+      .simulate('input', { inputType: 'deleteContentBackward' });
+    assert.equal(wrapper.find('datalist option').length, 0);
+  });
 
-    beforeEach(function() {
-      onEditTags = sinon.stub();
-      element = util.createDirective(document, 'tag-editor', {
-        onEditTags: { args: ['tags'], callback: onEditTags },
-        tags: ['foo'],
-      });
-      element.ctrl.onTagsChanged();
+  it('does not clear the suggestions when deleting only part of the input', () => {
+    const wrapper = createComponent();
+    wrapper.find('input').instance().value = 'tag3';
+    typeInput(wrapper);
+    assert.equal(wrapper.find('datalist option').length, 2);
+    wrapper.find('input').instance().value = 't'; // non-empty input remains
+    wrapper.find('input').simulate('keyup', { key: 'deleteContentBackward' });
```

Comment:
`deleteContentBackward` is a value of the [`InputEvent.inputType`](https://developer.mozilla.org/en-US/docs/Web/API/InputEvent/inputType) property rather than [`KeyEvent.key`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values)

Reply:
That was a mistake, intended to change that too